addons/purchase_stock/models/stock_move.py

```diff
@@ -209,7 +209,7 @@ def _get_value_from_quotation(self, quantity, at_date=None):
         # TODO: Start from global value
         if not self.purchase_line_id:
             return super()._get_value_from_quotation(quantity, at_date)
-        price_unit = self.purchase_line_id._get_stock_move_price_unit()
+        price_unit = self.purchase_line_id.with_context(move_id=self)._get_stock_move_price_unit()
```

Comment:
```suggestion
        price_unit = self.purchase_line_id.with_context(stock_move=self)._get_stock_move_price_unit()
```
`move_id` could suggest this only contains `self.id`